src/main/webapp/dev/js/main/adminHome.es6

```diff
@@ -1,14 +1,149 @@
+ */
+class InstructorAjaxResult {
+    constructor(instructor, responseData) {
+        assertDefined(instructor,
+                'InstructorAjaxResult cannot be constructed without instructor being defined.');
+        assert(!instructor.isError() || responseData === null,
+                'Response data must be null if instantiating with an error.');
+        this.instructor = instructor;
+        this.responseData = responseData;
+    }
+    /* eslint-disable class-methods-use-this */
+    isError() {
+        return this.instructor.isError();
+    }
+    /* eslint-enable class-methods-use-this */
+    isAddFailed() {
+        return !this.isError() && !this.responseData.isInstructorAddingResultForAjax;
+    }
+    /**
+     * If The AJAX request never happened because the instructor itself is erroneous,
+     * return the error message encapsulated in the InstructorError.
+     * Otherwise, return the AJAX status message.
+     */
+    getStatusMessage() {
+        if (this.isError()) {
+            return this.instructor.getErrorMessage();
+        }
+        return this.responseData.statusForAjax;
+    }
+}
+
 /**
- * Functions defined and used in `/adminHome`
+ * Stands in for an Instructor whenever an error occurs.
+ * The isError() method returns true here (but false in the Instructor class).
+ * In the event that a string was used to construct the Instructor
+ * but the construction failed, that string can be passed in to the InstructorError constructor.
  */
+class InstructorError {
+    constructor(message, originalString) {
+        this.message = message;
+        this.originalString = originalString;
+    }
+    toString() {
+        return this.originalString;
+    }
+    /* eslint-disable class-methods-use-this */
+    isError() {
+        return true;
+    }
+    /* eslint-enable class-methods-use-this */
+    getErrorMessage() {
+        return this.message;
+    }
+}
+
+/**
+ * Represents an instructor.
+ * Contains the shortname, name, email and institution fields.
+ * Static "constructor" functions are used since constructors cannot be overloaded in ES6.
+ */
+class Instructor {
+    /**
+     * Takes in several instructor attributes and constructs an instructor.
+     */
+    static create(shortName, name, email, institution) {
+        const instructor = new Instructor();
+        instructor.shortName = shortName;
+        instructor.name = name;
+        instructor.email = email;
+        instructor.institution = institution;
+        return instructor;
+    }
+    /**
+     * Takes in a string in either of the following formats:
+     *  NAME | EMAIL | INSTITUTION
+     *  NAME\tEMAIL\tINSTITUTION
+     * If the string is in the correct format, parses it and constructs an Instructor.
+     * If the format is wrong, constructs and returns a InstructorError instead.
+     */
+    static createFromString(str) {
+        const regexForTestingStringFormat = /^([^|\t]+)(?:\t|(?:\s\|\s))([^|\t]+)(?:\t|(?:\s\|\s))([^|\t]+)$/;
```

Comment:
Can this be written in a readable manner, eg. extract `([^|\t]+)` and `(?:\t|(?:\s\|\s))` as variables?